iceoryx2-cli/iox2/src/commands.rs

```diff
@@ -0,0 +1,160 @@
+use colored::*;
+use std::env;
+use std::fs;
+use std::path::PathBuf;
+use std::process::{Command, Stdio};
+use thiserror::Error;
+
+#[derive(Clone, Debug)]
+struct CommandInfo {
+    name: String,
+    path: PathBuf,
+    is_development: bool,
+}
+
+#[derive(Error, Debug)]
+pub enum ExecutionError {
+    #[error("Command not found: {0}")]
+    NotFound(String),
+    #[error("Execution failed: {0}")]
+    Failed(String),
+}
+
+pub fn list() {
+    println!("Installed Commands:");
+    let installed_commands = find();
+    for command in installed_commands {
+        println!(
+            "  {}",
+            format!(
+                "{}{}",
+                if command.is_development { "(dev) " } else { "" },
+                command.name.bold(),
+            )
+        );
+    }
+}
+
+fn find() -> Vec<CommandInfo> {
+    let development_commands = find_command_binaries_in_development_dirs();
+    let installed_commands = find_command_binaries_in_system_path();
+
+    let mut all_commands = development_commands;
+    all_commands.extend(installed_commands.iter().cloned());
+    all_commands
+}
+
+fn find_command_binaries_in_development_dirs() -> Vec<CommandInfo> {
+    let mut commands = Vec::new();
+    let current_exe = match env::current_exe() {
+        Ok(exe) => exe,
+        Err(_) => return commands,
+    };
+    let build_type = if cfg!(debug_assertions) {
+        "debug"
+    } else {
+        "release"
+    };
+
+    // Get the location of the binary directory for the build
+    let binary_dir = current_exe
+        .parent()
+        .unwrap()
+        .parent()
+        .unwrap()
+        .join(build_type);
+
+    if let Ok(entries) = fs::read_dir(&binary_dir) {
+        for entry in entries.filter_map(Result::ok) {
+            let path = entry.path();
+            if is_valid_command_binary(&path) {
+                if let Some(command_name) = path.file_name().and_then(|n| n.to_str()) {
+                    let stripped = command_name.strip_prefix("iox2-").unwrap_or(command_name);
+                    commands.push(CommandInfo {
+                        name: stripped.to_string(),
+                        path,
+                        is_development: true,
+                    });
+                }
+            }
+        }
+    }
+
+    commands
+}
+
+fn find_command_binaries_in_system_path() -> Vec<CommandInfo> {
```

Comment:
@orecham Since you are looking for binaries in multiple places, could you maybe add a command that lists all directories that you are checking for binaries?

Reply:
@elfenpiff Added.